aiida/calculations/transfer.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Implementation of CalcJobNode to add two numbers for testing and demonstration purposes."""
+import os
+from aiida import orm
+from aiida.engine import CalcJob
+from aiida.common.datastructures import CalcInfo
+
+
+def validate_transfer_inputs(inputs, ctx):  # pylint: disable=inconsistent-return-statements, unused-argument
+    """Validate the inputs"""
+    source_nodes = inputs['source_nodes']
+    instructions = inputs['instructions']
+
+    if not 'retrieve_files' in instructions.get_dict():
```

Comment:
you call `instructions.get_dict()` again on line 31. Better call it once and assign to variable